api/envoy/api/v2/route/route.proto

```diff
@@ -213,11 +229,15 @@ message WeightedCluster {
     // :ref:`envoy_api_msg_RouteConfiguration`. For more information, including details on
     // header value syntax, see the documentation on :ref:`custom request headers
     // <config_http_conn_man_headers_custom_request_headers>`.
-    repeated core.HeaderValueOption request_headers_to_add = 4;
+    repeated core.HeaderValueOption request_headers_to_add = 4
+        [(validate.rules).repeated .max_items = 1000];
 
     // Specifies a list of HTTP headers that should be removed from each request when
     // this cluster is selected through the enclosing :ref:`envoy_api_msg_route.RouteAction`.
-    repeated string request_headers_to_remove = 9;
+    repeated string request_headers_to_remove = 9 [
+      (validate.rules).repeated .items.string.max_bytes = 16384,
+      (validate.rules).repeated .max_items = 1000
```

Comment:
If you want to keep it < 2^32, this could end up being fragile, as any new header list field could take you over the limit. E.g. if we have 17 * 1K headers * 16KB per header.

Reply:
That's my concern... But haven't looked at the new limits.

Reply:
I was being dense; it's actually safe I think, I was reading 16kb as exponent 16 rather than 14 😴

Reply:
I think we're safe as a result, since it will be unlikely to add a 100 fields like this..